src/Simulation/Simulators/SparseSimulator/Native/types.h

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <vector>
+#include <complex>
+#include <unordered_map>
+#include <bitset>
+#include <shared_mutex>
+#include <mutex>
+
+#include "quantum_hash_map.hpp"
+
+namespace Microsoft::Quantum::SPARSESIMULATOR
+{
+
+using mutex_type = std::mutex;
+using recursive_mutex_type = std::recursive_mutex;
+using lock_type = std::lock_guard<mutex_type>;
+using recursive_lock_type = std::lock_guard<recursive_mutex_type>;
```

Comment:
Never used?

Reply:
It was used for multithreading, but now I removed multithreading and these types.